### PR TITLE
Fix #59: SWD update to attached nRF52811

### LINKED ISSUE #59: SWD update to attached nRF52811

### PR DESCRIPTION
**Pull request by TheSomeMan**
Before merging need to compile new firmware version for nRF52 with the latest ruuvi.boards.c submodule.

### CODE REVIEW

**Comment by ojousima** [comment]
Thanks, I'll compile 52811 with updated submodule latest by tomorrow

**Comment by ojousima** [comment]
I cleared out false positive bugs on SonarClouds and downgraded some issues such as dynamic memory use from status of bug to status of code smell

**Comment by ojousima** [comment]
Works on A1 with https://github.com/ruuvi/ruuvi.gateway_nrf.c/releases/tag/v0.5.0 which has updated board definitions

**Comment by ojousima** [comment]
retest this please

**Review by ojousima — APPROVED** [review]
Thanks, looks great. A2 hardware is not ready yet, so we cannot run this on actual hardware. This can be merged and any quality issues worked on while waiting for the new HW round

**Comment by TheSomeMan** [comment]
> I cleared out false positive bugs on SonarClouds and downgraded some issues such as dynamic memory use from status of bug to status of code smell

thanks, I've also performed refactoring and fixed many code smells

> adv_post_task(void *arg) should have a return type of BaseType_t if I'm understanding correctly?

adv_post_task does not relate to the current task, let's fix it as a part of #101

> Works on A1 with https://github.com/ruuvi/ruuvi.gateway_nrf.c/releases/tag/v0.5.0 which has updated board definitions

I'll update reference to nRF52 firmware

File: main/nrf52swd.c [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+
+#define NRF52_NVMC_REG_CONFIG__WEN_MASK (0x03UL)
+#define NRF52_NVMC_REG_CONFIG__WEN_REN  (0U)
+#define NRF52_NVMC_REG_CONFIG__WEN_WEN  (1U)
+#define NRF52_NVMC_REG_CONFIG__WEN_EEN  (2U)
+
+#define NRF52_NVMC_REG_ERASEALL__NO_OP (0U)
+#define NRF52_NVMC_REG_ERASEALL__ERASE (1U)
+
+#define NRF52_GPIO_SWD_CLK GPIO_NUM_15
```
Comment (ojousima) [inline main/nrf52swd.c:36]: Please move these definitions to ruuvi.boards.c

**Comment by ojousima** [comment]
As a note, the gateway had crashed overnight. Power cycle fixed the problem. I'm not sure if this is related to this pull request or if I just discovered some earlier problem
image

**Comment by TheSomeMan** [comment]
> As a note, the gateway had crashed overnight. Power cycle fixed the problem. I'm not sure if this is related to this pull request or if I just discovered some earlier problem

yes, I've also faced with this problem, but it does not relate to swd and firmware upgrading subsystems. I tried to comment calling of 'nrf52fw_update_fw_if_necessary' in the main, but gateway has still continued to crash. So, I believe the problem is caused by recent mergers with the master branch. I'm going to find the reason of this problem.

**Review by ojousima — APPROVED** [review]
Thanks

**Pull request merged**